### PR TITLE
async calculate build cache size

### PR DESCRIPTION
**Pull request by live1206**

### CODE REVIEW

File: src/observers/telemetryObserver.ts [hunk 1]
```diff
@@ -34,8 +37,8 @@ export class TelemetryObserver {
             case EventType.BuildCompleted:
                 this.handleBuildCompleted(<BuildCompleted>event);
                 break;
-            case EventType.BuildCacheSizeCalculated:
-                this.handleBuildCacheSize(<BuildCacheSizeCalculated>event);
+            case EventType.CalculateBuildCacheSize:
```
Comment (928PJY) [inline src/observers/telemetryObserver.ts]: Can we reuse the `DocfxRestoreCompleted`?

**Comment by 928PJY** [comment]
>         sinon.stub(utils, 'getFolderSizeInMB').resolves(10);

then remove this mock? and you can use this mock in `telemetryObserver.test.ts`

---
Refers to: test/unitTests/build/buildExecutor.test.ts:51 in 4db8d7e. [](commit_id = 4db8d7eb5162b87cce3298bcf0daaf1dfcc9544f, deletion_comment = False)

File: src/observers/telemetryObserver.ts (hunk 1 above)
Comment (live1206) [reply src/observers/telemetryObserver.ts]: I think we should use `DocfxRestoreStarted`

**Review by 928PJY — APPROVED** [review]
:shipit:

**Pull request merged**